library/n2_protocol.hpp

```diff
@@ -3,37 +3,31 @@
 #include "elliptics/interface.h"
 #include "elliptics/utils.hpp"
 
-// Base class for requests and responses
-struct n2_message {
+// Base class for requests
+struct n2_request {
+	// Adopted for broadcast sending: for each send cmd is modified, but body is constant.
 	dnet_cmd cmd;
-	
-	explicit n2_message(const dnet_cmd &cmd);
-	virtual ~n2_message() = default;
-};
-
-// Base class for requests. Responses are inherited directly from n2_message.
-struct n2_request : n2_message {
 	dnet_time deadline;
-	
+	std::shared_ptr<void> body;
```

Comment:
Looks like it is unused, isn't it?

Reply:
As I understand, body is unused only for `lookup` request.

Reply:
@artsel is right. There are many requests with body in elliptics.